tools/souffle_converter.py

```diff
@@ -1046,14 +1047,87 @@ def convert_terms(self, terms):
                 result.append(term)
         return result + tail
 
+    def flatten(self, tail, listbody, bool_dis):
+        for x in tail.children:
+            if (x.symbol.name == "Term"):
+                if (getOptField(x, "Literal") != None) or (getOptField(x, "!") != None):
+                    listbody.append(x)
+                child = getOptField(x, "Disjunction")
+                if child != None:
+                    if getOptField(child, "OR") != None:
+                        bool_dis = True
+                    listbody.append(child)
+            elif (x.symbol.name in ["OR", "(", ")", ",", "!"]):
+                listbody.append(x)
+            else:
+                (_, bool_dis) = self.flatten(x, listbody, bool_dis)
+        return (listbody, bool_dis)
+
+    def findCommas(self, comma):
+        commas = False
+        for x in comma.children:
+            if x.symbol.name == ",":
+                return True
+            elif not (x.symbol.name in ["Term", "OR", "(", ")", "!"]):
+                commas = self.findCommas(x)
```

Comment:
I think that this would be slightly more readable using no "commas" variable and returning directly here, and False below.